tests/system_tests_http.py

```diff
@@ -315,13 +315,10 @@ def address():
         self.assert_get("https://localhost:%s" % r.ports[0])
         # requireSsl=false Allows simple-ssl HTTP
 
-        # DISPATCH-1513: libwebsockets versions 3.2.0 introduces a new flag called
+        # DISPATCH-1513: libwebsockets version 3.2.0 introduced a new flag called
         # LWS_SERVER_OPTION_ALLOW_HTTP_ON_HTTPS_LISTENER
         # The new flag allows (as the flag says) HTTP over HTTPS listeners.
-        # Since this flag is not available before lws 3.2.0 we need
-        # to selectively disable this check
-        if skupper_router_site.LIBWEBSOCKETS_VERSION >= (3, 2, 0):
-            self.assert_get("http://localhost:%s" % r.ports[0])
+        self.assert_get("http://localhost:%s" % r.ports[0])
```

Comment:
the feature seems a bit unsafe, thinking about it; like, https downgrade attack?

Reply:
This should be mentioned to @grs for definitive judgement.

Reply:
There is a requireSsl flag on the listener entity as seen here - https://github.com/skupperproject/skupper-router/blob/main/python/skupper_router/management/skrouter.json#L731

I think this code is trying to honor the value of that flag. If requireSsl is set to false, then a we are allowed to make a http request over https.

This begs the question, why would someone associate an sslProfile with a listener and then go out of the way to set requireSsl to false ? I don't know the exact answer but see here on the amqp side - https://github.com/skupperproject/skupper-router/blob/main/src/server.c#L416

On the bright side, the adaptors don't have such a flag. In the case of adaptors, if an sslProfile is associated with listener, then ssl is required by default.

Reply:
I think we could get rid of the option, I agree it seems likely to result in unsafe configuration if used.

Reply:
https://github.com/skupperproject/skupper-router/issues/572